docsrc/exts/sphinxlocal/builders/insertdatestamp.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+    sphinxlocal.builders.insertdatestamp
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Inserts a git datestamp into the context as 'gitstamp',
+    to make it available for template use.
+
+    Adds itself as a page context handler: gets invoked after source
+    is read but before html is output.
+
+    :version: 0.1
+    :author: Nicola Nye <nicolan@fastmail.com>
+
+    :copyright: Copyright 2007-2016 by the Cyrus team,
+    :license: BSD, see LICENSE for details.
+"""
+
+import git, datetime
+
+# Gets the datestamp of the latest commit on the given file
+# Converts the datestamp into something more readable
+# Skips files not known to git or those whose datestamp we can't parse.
+# Expected git datestamp format: 2017-06-07 11:57:38 +1000
+# Output to June 7, 2017
+def page_context_handler(app, pagename, templatename, context, doctree):
+        g = git.Git('.')
+        try:
+            updated = g.log('--pretty=format:%ai','-n 1',"%s.rst" % pagename)
+            context['gitstamp'] = datetime.datetime.strptime(updated, "%Y-%m-%d %H:%M:%S %z").strftime("%b %d %Y")
```

Comment:
I'm not very familiar with Jinja2, but if possible it would be better to pass a `datetime.datetime` instance into the context so that the template can decide how to format it.